pdns/pdns_recursor.cc

```diff
@@ -3603,16 +3603,18 @@ try
   setupNODThread();
 #endif /* NOD_ENABLED */
   
-  try {
-    if(!::arg()["lua-dns-script"].empty()) {
-      t_pdl = std::make_shared<RecursorLua4>();
-      t_pdl->loadFile(::arg()["lua-dns-script"]);
-      g_log<<Logger::Warning<<"Loaded 'lua' script from '"<<::arg()["lua-dns-script"]<<"'"<<endl;
+  if(worker) {
```

Comment:
I was wondering if you want to load this script on the distributor thread when query distribution is enabled. If not, you may want to update the condition to something like `if (worker && (!g_weDistributeQueries || t_id != s_distributorThreadID)) {`.

Reply:
yes! pushed!